test/integration/middleware/essence/CollectSealedAuctionMw.t.sol

```diff
@@ -0,0 +1,637 @@
+        vm.stopPrank();
+
+        vm.warp(100001);
+
+        vm.expectRevert("ENDED");
+        vm.prank(lila);
+        collectSealedAuctionMw.placeBid(bobbyProfileId, bobbyEssenceId, 22);
+
+        // lila collecting
+        vm.startPrank(lila);
+        uint256 collectId1 = link3Profile.collect(
+            DataTypes.CollectParams(lila, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+
+        vm.expectRevert("COLLECTOR_NO_WINS");
+
+        link3Profile.collect(
+            DataTypes.CollectParams(lila, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+
+        vm.stopPrank();
+
+        vm.startPrank(dave);
+        uint256 collectId2 = link3Profile.collect(
+            DataTypes.CollectParams(dave, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+        uint256 collectId3 = link3Profile.collect(
+            DataTypes.CollectParams(dave, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+        vm.stopPrank();
+
+        vm.startPrank(shane);
+        uint256 collectId4 = link3Profile.collect(
+            DataTypes.CollectParams(shane, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+        vm.stopPrank();
+
+        vm.prank(meera);
+        vm.expectRevert("COLLECTOR_NO_WINS");
+        link3Profile.collect(
+            DataTypes.CollectParams(meera, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+
+        vm.startPrank(bunty);
+        uint256 collectId5 = link3Profile.collect(
+            DataTypes.CollectParams(bunty, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+        vm.expectRevert("COLLECT_LIMIT_EXCEEDED");
+        link3Profile.collect(
+            DataTypes.CollectParams(bunty, bobbyProfileId, bobbyEssenceId),
+            new bytes(0),
+            new bytes(0)
+        );
+        vm.stopPrank();
+
+        bobbyEssNFT = link3Profile.getEssenceNFT(
+            bobbyProfileId,
+            bobbyEssenceId
+        );
+
+        assertEq(EssenceNFT(bobbyEssNFT).ownerOf(collectId1), lila);
+        assertEq(EssenceNFT(bobbyEssNFT).ownerOf(collectId2), dave);
+        assertEq(EssenceNFT(bobbyEssNFT).ownerOf(collectId3), dave);
+        assertEq(EssenceNFT(bobbyEssNFT).ownerOf(collectId4), shane);
+        assertEq(EssenceNFT(bobbyEssNFT).ownerOf(collectId5), bunty);
+        assertEq(IERC20(address(token)).balanceOf(meera), startingMeera);
+        assertEq(IERC20(token).balanceOf(shane), startingShane - 7);
+        assertEq(IERC20(token).balanceOf(dave), startingDave - 15);
+        assertEq(IERC20(token).balanceOf(bunty), startingBunty - 8);
+        // assertEq(IERC20(token).balanceOf(lila), startingLila-20);
```

Comment:
why commented?

Reply:
commented cause stack too deep error. Also a redundant check. If shane ,dave,bunty,meera , bobby and the contract has correct balances then lila automatically has correct balance.